packages/services/src/view-service/transaction-planner/fees.ts

```diff
@@ -0,0 +1,34 @@
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+
+// Attempts to extract a fee token from the assets used in the actions of the planner request
+// Priority in descending order
+export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+  const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
+  if (outputAsset) return outputAsset;
```

Comment:
Performance note: this will loop over the array twice, first to get the asset IDs, and second to find a truthy one. It'd be more efficient to rewrite it like this:

```suggestion
  const outputWithAssetId = request.outputs.find(o => o.value?.assetId);
  if (outputWithAssetId) return outputWithAssetId.value!.assetId;
```

(Same note for the instances below this as well.)

Reply:
the signature of the return type will `AssetId | undefined` in that case, and we shouldn't assume the requestor is guaranteed to send an `assetId` along with the request.